hypershift-operator/controllers/hostedcluster/manifests/manifests.go

```diff
@@ -67,6 +67,27 @@ func (o PullSecret) Build() *corev1.Secret {
 	return secret
 }
 
+type SigningKey struct {
```

Comment:
I've been slowly migrating this pattern to something that I think is more testable and useable across contexts — maybe we could follow it for all new resources as well. The pattern applied here would be:

1. Instead of this struct, declare a locator function to enable finding nodes in our managed graph without having to supply inputs useful only for reconciliation:
```go
func SigningKeyName(hostedClusterNamespace string) types.NamespacedName
```
2. Instead of a builder struct which really is only used during reconciliation, declare the desired state/reconcile logic inline in the `controllerutil.CreateOrUpdate` call (which is able to internally switch on DeletionTimestamp as necessary to account for defaulting for new resources vs. applying an update)
3. If that logic is big enough to warrant breaking out for unit testing, instead of declaring it inline, extract out a function (e.g. `reconcileHostedControlPlane`) which can be easily tested in isolation with table tests which would cover all defaulting and reconcile logic

Reply:
I was just blending in with the current code.  Seems like there are all kinds of ways to do this, even within our own code.

I didn't quite follow but... since this is how the surrounding code works at the time of this PR, can this be acceptable?

Reply:
I'll refactor after this, thanks!